source/gamemodes/BoostMod.cpp

```diff
@@ -0,0 +1,138 @@
+// gamemodes/BoostMod.cpp
+// A general team based boost modifier gamemode for Rocket Plugin.
+// Credits to Stanbroek (rocket plugin author) 
+// and ubelhj (author of SpeedBoost plugin, source of some of the code here).
+// 
+// Author:       Al12
+// Version:      0.1.0 17/07/20
+// BMSDKversion: 95
+
+#include "BoostMod.h"
+
+
+/// <summary>Checks and updates boost every game tick.</summary>
+/// <remarks>Gets called on 'Function GameEvent_Soccar_TA.Active.Tick'.</remarks>
+/// <param name="server"><see cref="ServerWrapper"/> instance of the current server</param>
+/// <param name="params">Delay since last update</param>
+void BoostMod::onTick(ServerWrapper server, void*)
+{
+	if (server.IsNull()) {
+		return;
+	}
+
+	float maxBoost = 1.0f;
+
+	if (generalBoostSettings.enabled) {
+		maxBoost = (float)generalBoostSettings.maxBoost / 100.0f;
+	}
+
+	auto teams = server.GetTeams();
+
+	for (int teamIdx = 0; teamIdx < teams.Count(); teamIdx++) {
+		float currentTeamMax = maxBoost;
+
+		if (teamIdx < teamsBoostSettings.size() && teamsBoostSettings[teamIdx].enabled) {
+			currentTeamMax = (float)teamsBoostSettings[teamIdx].maxBoost / 100.0f;
+		}
+		
+		if (generalBoostSettings.enabled || teamsBoostSettings[teamIdx].enabled) {
```

Comment:
You check if the `teamIdx` is a valid index for `teamsBoostSettings`, but if it is not you still try to access it.